src/expm.jl

```diff
@@ -25,25 +26,50 @@ end
 
     v = (a-d)^2 + 4*b*c
 
+    m = (a + d) / 2
+
     if v > 0
-        z = sqrt(v)
-        z1 = cosh(z / 2)
-        z2 = sinh(z / 2) / z
+        # In this case the formulas in the entries of the matrix
+        # are a function of cosh and sinh, and could (in theory)
+        # follow the same code pattern of the other branches (v ≤ 0).
+        # However, cosh and sinh explode with large arguments and
+        # we use the following identity to avoid numerical issues:
+        #
+        # exp(m) * [c₁ * cohs(δ) + c₂ * sinh(δ)] =
+        #   c₁ * (e₊ + e₋) / 2 + c₂ * (e₊ - e₋) / 2
+        #
+        # where e₊ = exp(m + δ) and e₋ = exp(m - δ).
+        #
+        # See https://github.com/JuliaArrays/StaticArrays.jl/issues/1295
+        δ = sqrt(v) / 2
+        e₊ = exp(m + δ)
+        e₋ = exp(m - δ)
+        e₁ = e₊ + e₋
+        e₂ = e₊ - e₋
+        c₂ = (a - d) / 2δ
+        m11 = (e₁ + c₂ * e₂) / 2
+        m12 = (b / δ) * e₂ / 2
+        m21 = (c / δ) * e₂ / 2
+        m22 = (e₁ - c₂ * e₂) / 2
```

Comment:
```suggestion
        e₊ = exp(m + δ) / 2
        e₋ = exp(m - δ) / 2
        e₁ = e₊ + e₋
        e₂ = e₊ - e₋
        c₂ = (a - d) / 2δ
        m11 = e₁ + c₂ * e₂
        m12 = (b / δ) * e₂
        m21 = (c / δ) * e₂
        m22 = e₁ - c₂ * e₂
```
Looks like this might save a couple multiplies, but really it was fine the way it was too.

Reply:
Thank you @mikmoore , I will update the code accordingly.